### PR TITLE
feat: changed the behaviour of DND button

### PR DESCRIPTION
**Pull request by Mayank4352**
Made changes as requested in #66 except confetti

### CODE REVIEW

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 1]
```diff
@@ -42,11 +44,13 @@ class TipsOrganism extends StatelessWidget {
...
-                ? ButtonAtom(
-                    variant: ButtonVariant.primary,
-                    onPressed: () => DndController.instance.toggleDnd(),
-                    text: 'dnd',
-                  )
+                ? dndEnable
+                    ? TextAtom('dnd_on')
+                    : ButtonAtom(
+                        variant: ButtonVariant.primary,
+                        onPressed: () => DndController.instance.toggleDnd(),
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:51]: Can be simplified to 
                        ` onPressed:DndController.instance.toggleDnd,`

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 2]
```diff
@@ -42,11 +44,13 @@ class TipsOrganism extends StatelessWidget {
...
+                        text: 'Do Not Disturb',
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:52]: Please support translations

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 3]
```diff
@@ -42,11 +44,13 @@ class TipsOrganism extends StatelessWidget {
...
+                      )
                 : const TextAtom('dnd_ios'),
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:54]: If we can dnd state on iOS the text should also view 'dnd_on' state

File: lib/presentation/pages/intro_page.dart [hunk 4]
```diff
@@ -19,6 +20,18 @@ class _IntroPageState extends State<IntroPage> {
...
+  bool? dndEnabled;
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      bool dnd = await FlutterDnd.getCurrentInterruptionFilter() ==
+          FlutterDnd.INTERRUPTION_FILTER_PRIORITY;
+      setState(() {
+        dnd ? dndEnabled = true : dndEnabled = false;
+      });
+    });
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:32]: In initState it is recommended to put the code after `super.initState();`
Also I prefer to not use `addPostFrameCallback` and instead show a loading widget if needed.

Also this code block should move to a separate function

File: lib/presentation/pages/intro_page.dart [hunk 5]
```diff
@@ -68,7 +81,7 @@ class _IntroPageState extends State<IntroPage> {
           PageViewModel(
             titleWidget: const SizedBox(),
             decoration: emptyPageDecoration(),
-            bodyWidget: TipsOrganism(studyType),
+            bodyWidget: TipsOrganism(studyType, dndEnabled ?? false),
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:84]: I understand where this change coming from but it is better to keep the code responsible for the page and not passing it on (maybe if it is long operations which it is not).

We might also change the check dnd to a listener in the future to change the state live while the user toggle it from the settings

**Review by guyluz11 — CHANGES_REQUESTED** [review]
It is nice to see a new face :), welcome

File: lib/presentation/organisms/intro/tips_organism.dart (hunk 3 above)
Comment (Mayank4352) [reply lib/presentation/organisms/intro/tips_organism.dart:54]: Currently we cannot detect the state of dnd on ios [as mentioned in this answer](https://stackoverflow.com/a/17560797)

Comment (guyluz11) [reply lib/presentation/organisms/intro/tips_organism.dart:54]: Amazing Apple.
Ok thanks for the comment

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 6]
```diff
@@ -2,21 +2,43 @@ import 'dart:io';
 
 import 'package:easy_localization/easy_localization.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_dnd/flutter_dnd.dart';
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:5]: All use of the package should stay inside of the dnd_repository.dart

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 7]
```diff
@@ -2,21 +2,43 @@ import 'dart:io';
...
 
+  @override
+  void initState() {
+    super.initState();
+    setDnd();
+  }
+
+  Future<void> setDnd() async {
+    final bool dnd = await FlutterDnd.getCurrentInterruptionFilter() ==
+        FlutterDnd.INTERRUPTION_FILTER_PRIORITY;
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:35]: Move to dnd_repository.dart as a function

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 8]
```diff
@@ -2,21 +2,43 @@ import 'dart:io';
...
+
+    setState(() {
+      dnd ? dndEnable = true : dndEnable = false;
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:38]: `dndEnable = dnd;` ?

File: lib/presentation/organisms/intro/tips_organism.dart [hunk 9]
```diff
@@ -29,49 +51,58 @@ class TipsOrganism extends StatelessWidget {
...
-                    text: 'dnd',
-                  )
-                : const TextAtom('dnd_ios'),
-          ),
-          ListView.builder(
-            shrinkWrap: true,
-            itemBuilder: (BuildContext context, int index) {
-              final Tip tip = beforeStudyTips[index];
+    return dndEnable == null
+        ? Center(child: CircularProgressIndicator())
```
Comment (guyluz11) [inline lib/presentation/organisms/intro/tips_organism.dart:55]: Please create atom for CircularProgressIndicator

File: lib/infrastructure/dnd_repository.dart [hunk 10]
```diff
@@ -3,14 +3,10 @@ part of 'package:infinite_horizons/domain/dnd_controller.dart';
 class _DndRepository extends DndController {
   @override
   Future<void> toggleDnd() async {
```
Comment (guyluz11) [inline lib/infrastructure/dnd_repository.dart:5]: This function does not toggle anymore, should change to enableDnd or something

File: lib/presentation/organisms/intro/tips_organism.dart (hunk 9 above)
Comment (guyluz11) [reply lib/presentation/organisms/intro/tips_organism.dart:55]: Also let's change it to a nicer version
if(dndEnable == null){
    return Center(child: CircularProgressIndicator());
}

return MarginedExpandedAtom(...

File: lib/presentation/pages/intro_page.dart [hunk 11]
```diff
@@ -1,4 +1,5 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_dnd/flutter_dnd.dart';
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:2]: Redundant

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by guyluz11** [comment]
Hey @Mayank4352 

Sorry of the inconvenience but I have thought of a better design for the DND button and all this page in general.

image



Let me know if you still up for the task.

**Comment by guyluz11** [comment]
I see that you have closed the pr and didn't respond so I will take it as your answer is no and will start work on it.

Thanks and sorry for changing the requirements in the middle

**Comment by M4dhav** [comment]
@guyluz11 He's still working on it, the PR was closed when he tried to resolve conflicts by discarding commits on the branch

**Comment by guyluz11** [comment]
> @guyluz11 He's still working on it, the PR was closed when he tried to resolve conflicts by discarding commits on the branch

I asked "Let me know if you are still up for the task." and he didn't respond, so I continued on.
Communication is part of the process

**Pull request closed without merging**